packages/runtime/container-runtime/src/summary/summarizerTypes.ts

```diff
@@ -320,10 +304,9 @@ export interface ISummarizerEvents extends IEvent {
 	(event: "summarizingError", listener: (error: ISummarizingWarning) => void);
 }
 
-export interface ISummarizer
-	extends IEventProvider<ISummarizerEvents>,
-		IFluidLoadable,
-		Partial<IProvideSummarizer> {
+export interface ISummarizer extends IEventProvider<ISummarizerEvents> {
+	readonly ISummarizer?: ISummarizer;
```

Comment:
Can we document this? What does it mean for this to be `undefined`?

Reply:
I actually don't think this *can* be undefined...? I'm adding a bit of docs to the best of my ability, and this property basically exists to support our `FluidObject<IProvideSomething>` "discovery" pattern. `IProvideSomething` is usually an interface that exposes _non-optional_ properties name `IProvideSomething` and/or `ISomething`, of **type** `ISomething` (I might be misremembering details here but the overall idea should be correct). When we use `FluidObject<>` to discover if some object implements/provides some functionality, the corresponding property there is optional because the object might not implement/provide the thing. But in the `IProvideSomething` / `ISomething` interface itself, I don't think there's a reason for it to be optional. So this one being optional might have been an oversight.

For reference, [the PR](https://github.com/microsoft/FluidFramework/pull/8299/files#diff-2fd2dbab531aa6628d4aa1d27ab0f2d0237967f42c4758d78b48bb038a239000) where `ISummarizer` started extending `Partial<IProvideSummarizer>`, which gave it this property as optional.